src/main/scala/mesosphere/marathon/core/launcher/impl/InstanceOpFactoryHelper.scala

```diff
@@ -42,6 +42,19 @@ class InstanceOpFactoryHelper(
     InstanceOp.LaunchTaskGroup(executorInfo, groupInfo, stateOp, oldInstance = None, createOperations)
   }
 
+  def provision(
+    taskInfo: Mesos.TaskInfo,
+    newTask: Task,
+    instance: Instance): InstanceOp.LaunchTask = {
+
+    assume(newTask.taskId.mesosTaskId == taskInfo.getTaskId, "marathon task id and mesos task id must be equal")
+
+    def createOperations = Seq(offerOperationFactory.launch(taskInfo))
+
+    val stateOp = InstanceUpdateOperation.Provision(instance)
+    InstanceOp.LaunchTask(taskInfo, stateOp, oldInstance = None, createOperations)
```

Comment:
I really hope to get rid of these operations completely. However, for the time being we have to live with them. @timcharper, how did you model actions/effects in your prototype?